diff --git a/dist/changesets-cli.esm.js b/dist/changesets-cli.esm.js
index 1e945455b39b9c6424b26960b3d26a095ad980e4..b4ad2a7f34c226b998d0deef0272655d73115577 100644
--- a/dist/changesets-cli.esm.js
+++ b/dist/changesets-cli.esm.js
@@ -222,16 +222,16 @@ async function confirmMajorRelease(pkgJSON) {
 async function getPackagesToRelease(changedPackages, allPackages) {
   function askInitialReleaseQuestion(defaultChoiceList) {
     return askCheckboxPlus(
-    // TODO: Make this wording better
-    // TODO: take objects and be fancy with matching
-    `Which packages would you like to include?`, defaultChoiceList, x => {
-      // this removes changed packages and unchanged packages from the list
-      // of packages shown after selection
-      if (Array.isArray(x)) {
-        return x.filter(x => x !== "changed packages" && x !== "unchanged packages").map(x => cyan(x)).join(", ");
-      }
-      return x;
-    });
+      // TODO: Make this wording better
+      // TODO: take objects and be fancy with matching
+      `Which packages would you like to include?`, defaultChoiceList, x => {
+        // this removes changed packages and unchanged packages from the list
+        // of packages shown after selection
+        if (Array.isArray(x)) {
+          return x.filter(x => x !== "changed packages" && x !== "unchanged packages").map(x => cyan(x)).join(", ");
+        }
+        return x;
+      });
   }
   if (allPackages.length > 1) {
     const unchangedPackagesNames = allPackages.map(({
@@ -596,9 +596,8 @@ async function getPublishTool(cwd) {
   const pm = await detect({
     cwd
   });
-  if (!pm || pm.name !== "pnpm") return {
-    name: "npm"
-  };
+  if (!pm) return { name: "npm" };
+  if (pm.name === "yarn") return { name: "yarn" };
   try {
     let result = await spawn$1("pnpm", ["--version"], {
       cwd
@@ -733,6 +732,9 @@ async function internalPublish(packageJson, opts, twoFactorState) {
   } = publishTool.name === "pnpm" ? await spawn$1("pnpm", ["publish", "--json", ...publishFlags], {
     env: Object.assign({}, process.env, envOverride),
     cwd: opts.cwd
+  }) : publishTool.name === "yarn" ? await spawn$1("yarn", ["npm", "publish", ...publishFlags], {
+    env: Object.assign({}, process.env, envOverride),
+    cwd: opts.cwd,
   }) : await spawn$1(publishTool.name, ["publish", opts.publishDir, "--json", ...publishFlags], {
     env: Object.assign({}, process.env, envOverride)
   });
@@ -1436,8 +1438,8 @@ ${format("", err).replace(process.cwd(), "<cwd>")}
 ## Versions
 
 - @changesets/cli@${
-    // eslint-disable-next-line import/no-extraneous-dependencies
-    require("@changesets/cli/package.json").version}
+      // eslint-disable-next-line import/no-extraneous-dependencies
+      require("@changesets/cli/package.json").version}
 - node@${process.version}
 
 ## Extra details
