diff --git a/dist/error.d.ts b/dist/error.d.ts
index bd657d5311ff3d255dc57a2f7224301d532a3177..8924e0982c64c566f4d9808cde62292d7ed334ac 100644
--- a/dist/error.d.ts
+++ b/dist/error.d.ts
@@ -1,9 +1,9 @@
 import { D as DiffOptions } from './types-widbdqe5.js';
 import 'pretty-format';
 
-declare function serializeError(val: any, seen?: WeakMap<WeakKey, any>): any;
+declare function serializeError(val: any, seen?: WeakMap<object, any>): any;
 declare function processError(err: any, diffOptions?: DiffOptions): any;
-declare function replaceAsymmetricMatcher(actual: any, expected: any, actualReplaced?: WeakSet<WeakKey>, expectedReplaced?: WeakSet<WeakKey>): {
+declare function replaceAsymmetricMatcher(actual: any, expected: any, actualReplaced?: WeakSet<object>, expectedReplaced?: WeakSet<object>): {
     replacedActual: any;
     replacedExpected: any;
 };
