diff --git a/lib/intercepted_request_router.js b/lib/intercepted_request_router.js
index d82c2446b540759cbc54ad7b26222a80109b8ec9..bf24add845d12e436c97b3ea5e2e38f87b77d489 100644
--- a/lib/intercepted_request_router.js
+++ b/lib/intercepted_request_router.js
@@ -68,7 +68,7 @@ class InterceptedRequestRouter {
     // For parity with Node, it's important the socket event is emitted before we begin playback.
     // This flag is set when playback is triggered if we haven't yet gotten the
     // socket event to indicate that playback should start as soon as it comes in.
-    this.readyToStartPlaybackOnSocketEvent = false
+    this.readyToStartPlaybackOnSocketEvent = true
 
     this.attachToReq()
 
